cassiopeia/data/CassiopeiaTree.py

```diff
@@ -162,12 +163,12 @@ def populate_tree(self, tree: Union[str, ete3.Tree, nx.DiGraph]) -> None:
         # add character states
         for n in self.nodes:
             if (
-                self.__original_character_matrix is not None
-                and n in self.__original_character_matrix.index.tolist()
+                self.__current_character_matrix is not None
+                and n in self.__current_character_matrix.index.tolist()
             ):
                 self.__network.nodes[n][
                     "character_states"
-                ] = self.__original_character_matrix.loc[n].to_list()
+                ] = self.__current_character_matrix.loc[n].to_list()
             else:
                 self.__network.nodes[n]["character_states"] = []
```

Comment:
The reason is twofold: 1) exactly what you mentioned about the sampling procedure. One thing that you may be mistaken is that the `SupercellularSampler` retains the original character matrix (with unambiguous cellular leaf names and states) and only modifies the current character matrix to contain merged states. When `populate_tree` is called on this tree, it tries to look up the old, cellular leaf states, which are now outdated.
I guess we could circumvent this by just setting the original & current character matrix to be the merged one, but either way I still think every operation should be done on the *current* object state rather than a (possibly) outdated original character matrix. My understanding of the `__original_character_matrix` is that it is like `adata.raw` where it exists if the user wants to recover the original version, but should not be used when operating on the anndata itself.